MDLIB/include/mdoodz.h

```diff
@@ -138,27 +140,29 @@ typedef struct {
 typedef double (*SetHorizontalVelocity_f)(MdoodzInput *input, Coordinates coordinates);
 typedef double (*SetVerticalVelocity_f)(MdoodzInput *input, Coordinates coordinates);
 typedef double (*SetTemperature_f)(MdoodzInput *input, Coordinates coordinates);
-typedef int (*SetPhase_f)(MdoodzInput *input, Coordinates coordinates);
-typedef int (*SetDual_f)(MdoodzInput *input, Coordinates coordinates, int phase);
+typedef int    (*SetPhase_f)(MdoodzInput *input, Coordinates coordinates);
+typedef int    (*SetDualPhase_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetGrainSize_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetPorosity_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetDensity_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetXComponent_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetPressure_f)(MdoodzInput *input, Coordinates coordinates, int phase);
 typedef double (*SetNoise_f)(MdoodzInput *input, Coordinates coordinates, int phase);
+typedef void   (*SetDefGrad_f)(double*, double*, double*, double*, MdoodzInput *input, Coordinates coordinates, int phase);
```

Comment:
Only thing is it would be nice to add names to new parameters, put those 4 arguments after the phase as a tensor argument

Reply:
What does SetDefGrad stands for?

Reply:
Very good suggestion! Now Tensor2D type is public and used for input function. 
DefGrad stands for Deformation Gradient. I think it's a sufficiently clear name. Anyone who wants to deal with finite strain computations can pick it up easily (especially once we will build up docs).